legacy/reqcounter/reqcounter_test.go

```diff
@@ -124,3 +126,57 @@ func TestRequestCounterIncrement(t *testing.T) {
 	// Ensure the request counter was incremented.
 	require.EqualValues(t, &expected, &requestCounter, "The request counter failed to increment its request field.")
 }
+
+func TestLog(t *testing.T) {
+	// Create multiple request counters with unique number of starting requests. These unique requests will
+	// be the way to differentiate between request counter log statements.
+	requestCounters := []rc.RequestCounter{
+		NewRequestCounter(21),
+		NewRequestCounter(9),
+		NewRequestCounter(2839),
+	}
+	netMonitor := &rc.NetworkMonitor{
+		RequestCounters: rc.RequestCounters{
+			&requestCounters[0],
+			&requestCounters[1],
+			&requestCounters[2],
+		},
+	}
+	// Mock time with a fake global time that remains constant. This avoids testing the order in which
+	// request counters are logged. For simplicity, additional tests may check whether a single request
+	// counter calls Debug() the expected number of times as fake time advances forward. This circumvents
+	// the task of concurrently communicating time-steps with multiple goroutines, demanding a
+	// complicated sleep and time-step synchronization.
```

Comment:
I think we still want to mock sleep otherwise we aren't testing the intended behavior.

I forget why modifying sleep to just do an infinite loop checking on currentFakeTime - rc.Since was running into issues?

Reply:
The issue with this is that all infinite loops do not synchronize together, as they are independent goroutines. This means, we can't rely on each counter to respond to time changes. The only way to do this is to use wait groups to stall the main reqcounter_test.go thread until all goroutines have "responded" to their new times.

Reply:
ACK